snippets/standard/datetime/json/datetime-converter-examples/example3/Program.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace DateTimeConverterExamples
+{
+    public class DateTimeConverterUsingDateTimeParseAsFallback : JsonConverter<DateTime>
+    {
+        public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw new JsonException();
```

Comment:
TryGetDateTime already validates that tokentype == string. Unless you are doing the check to throw a custom exception (rather than invalidoperationexception).